SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -564,6 +567,43 @@ void CMeshSolver::UpdateDualGrid(CGeometry *geometry, CConfig *config){
 
 }
 
+void CMeshSolver::ComputeGridVelocity_FromBoundary(CGeometry **geometry, CNumerics **numerics, CConfig *config){
+
+  /*--- Compute the stiffness matrix, no point recording because we clear the residual. ---*/
+
+  const bool wasActive = AD::BeginPassive();
+
+  Compute_StiffMatrix(geometry[MESH_0], numerics, config);
+
+  AD::EndPassive(wasActive);
+
+  /*--- Clear residual (loses AD info), we do not want an incremental solution. ---*/
+  SU2_OMP_PARALLEL {
+    LinSysRes.SetValZero();
+    LinSysSol.SetValZero();
+  }
+
+  /*--- Impose boundary conditions including bouundary velocity ---*/
+  SetBoundaryVelocities(geometry[MESH_0], numerics[FEA_TERM], config);
+
+  /*--- Solve the linear system. ---*/
+  Solve_System(geometry[MESH_0], config);
+  for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      su2double val_vel = LinSysSol(iPoint, iDim);
```

Comment:
It should also be valid to run this loop to nPoint and not communicate.

Reply:
Yes, I could switch to nPoint and remove the communication calls.